projects/core/src/occ/adapters/product/converters/product-name-normalizer.ts

```diff
@@ -1,25 +1,38 @@
 import { Injectable } from '@angular/core';
+import { Product } from '../../../../model/product.model';
+import { Converter } from '../../../../util/converter.service';
 import { OccConfig } from '../../../config/occ-config';
 import { Occ } from '../../../occ-models/occ.models';
-import { Converter } from '../../../../util/converter.service';
-import { Product } from '../../../../model/product.model';
 
 @Injectable({ providedIn: 'root' })
 export class ProductNameNormalizer implements Converter<Occ.Product, Product> {
   constructor(protected config: OccConfig) {}
 
   convert(source: Occ.Product, target?: Product): Product {
-    if (target === undefined) {
-      target = { ...(source as any) };
-    }
+    target = target ?? { ...((source as any) as Partial<Product>) };
+
     if (source.name) {
       target.name = this.normalize(source.name);
+      target.slug = this.normalizeSlug(source.name);
       target.nameHtml = source.name;
     }
-    return target;
+    return target as Product;
   }
 
   protected normalize(name: string): string {
     return name.replace(/<[^>]*>/g, '');
   }
+
+  /**
+   * Provides a title slug for the pretty URL.
+   *
+   * The name is created in lowercase, trimmed and all special characters that are candidates for url encoded
+   * are replaced by the slug char (dash by default).
+   */
+  protected normalizeSlug(name: string): string {
+    // https://developers.google.com/maps/documentation/urls/url-encoding
+    const reservedChars = ` !*'();:@&=+$,/?%#[]`;
+    const re = new RegExp(`[${reservedChars.split('').join('\\')}]`, 'g');
```

Comment:
Similar to the above - we could instantiate the `new RegExp` only once in this class, and avoid having to instantiate it every time the method is called.